packages/ripple-binary-codec/src/shamap.ts

```diff
@@ -3,13 +3,12 @@ import { HashPrefix } from './hash-prefixes'
 import { Sha512Half } from './hashes'
 import { Hash256 } from './types/hash-256'
 import { BytesList } from './serdes/binary-serializer'
-import { Buffer } from 'buffer/'
 
 /**
  * Abstract class describing a SHAMapNode
  */
 abstract class ShaMapNode {
-  abstract hashPrefix(): Buffer
+  abstract hashPrefix(): Uint8Array
   abstract isLeaf(): boolean
   abstract isInner(): boolean
   abstract toBytesSink(list: BytesList): void
```

Comment:
toBytesSink was never meant to have a dependency on a particular instance of a BytesSink ... BytesList should "satisifies"/"implements" BytesSink, simply `{ put(b: Uint8Array) }`

so, while you're at it :)

Reply:
So instead of having an abstract function here, just including a concrete one like:

```
toBytesSink(list: BytesList) {
  ...
}
```

(I don't really follow what you mean by `{ put(b: Uint8Array }` or that BytesList satisfies BytesSink unfortunately)